rerun_py/rerun_sdk/rerun/datatypes/mat3x3_ext.py

```diff
@@ -1,21 +1,60 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Sequence
 
 import numpy as np
-import numpy.typing as npt
+import pyarrow as pa
+
+from rerun.error_utils import _send_warning
 
 if TYPE_CHECKING:
-    from . import Mat3x3Like
+    from . import Mat3x3ArrayLike, Mat3x3Like
 
 
 class Mat3x3Ext:
+    def __init__(self: Any, rows: Mat3x3Like | None = None, *, columns: Mat3x3Like | None = None) -> None:
+        from . import Mat3x3
+
+        if rows is not None:
+            if columns is not None:
+                _send_warning("Can't specify both columns and rows of matrix.", 1, recording=None)
+
+            if isinstance(rows, Mat3x3):
+                self.flat_columns = rows.flat_columns
+            else:
+                arr = np.array(rows, dtype=np.float32).reshape(3, 3)
+                self.flat_columns = arr.flatten("F")
+        elif columns is not None:
+            arr = np.array(columns, dtype=np.float32).reshape(3, 3)
+            self.flat_columns = arr.flatten()
```

Comment:
Nit: this is a little bit of API abuse in my opinion so probably warrants a comment.

It's creating an array with columns as rows and then taking advantage of the fact that the default C order is the transpose of what the spec asks for.

Something like:

```suggestion
            arr = np.column_stack(np.array(columns, dtype=np.float32).reshape(3, 3))
            self.flat_columns = arr.flatten("F")
```
would be more explicit in my opinion.

Reply:
huh I find this actually more confusing, also it's more operations